tests/unit/anchore_engine/auth/test_common.py

```diff
@@ -0,0 +1,130 @@
+import pytest
+import json
+import time
+from anchore_engine.auth.common import get_docker_registry_userpw, get_creds_by_registry, registry_record_matches
+
+_test_username = 'tonystark'
+_test_password = 'potts'
+_test_registry_meta = {
+    'authorizationToken': '{}:{}'.format(_test_username, _test_password)
+}
+
+def test_get_docker_registry_userpw():
+    record_ecr = {
+        'registry_type': 'awsecr',
+        'registry_meta': json.dumps(_test_registry_meta),
+    }
+    record_ecr_bad_json = {
+        'registry_type': 'awsecr',
+        'registry_meta': 'this-is-not-valid-json!}',
+    }
+    record_not_ecr = {
+        'registry_type': 'other-registry',
+        'registry_user': _test_username,
+        'registry_pass': _test_password,
+    }
+
+    test_cases = [
```

Comment:
pytest has some nice annotations for parameterizing a test, useful in the future but non-blocking here. See as an example: https://github.com/anchore/anchore-engine/blob/master/tests/unit/anchore_engine/services/test_api_specs.py#L26

Reply:
Thanks @zhill — I went ahead and reworked this a bit, since it was a good excuse for me to learn more Python idioms. I checked out the example you mentioned, as well as the pytest docs on [parameterization](https://docs.pytest.org/en/latest/parametrize.html#pytest-mark-parametrize-parametrizing-test-functions) and [expected exceptions](https://docs.pytest.org/en/stable/assert.html#assertions-about-expected-exceptions).

Reply:
Great, looks good!